File: wrapped/funcs/f_70.py

import re

def is_spam(text):
    # Check for unusual numeric or special characters percentage
    non_alphabetic_chars = sum(not c.isalnum() for c in text)
    percentage = non_alphabetic_chars / len(text)
    if percentage > 0.3:
        return True

    # Check for excessively long alphanumeric strings (potential URLs)
    alphanumeric_chunks = re.compile(r'\S+').split(text)
    for chunk in alphanumeric_chunks:
        if len(chunk) > 20:
            return True

    # Check for common spam phrases
    spam_phrases = ['상한가', '최고이자율', '특별정보', 'M반도체', '적금', '출금', '출시', '이벤트', 
                    '공개', '혜택', '우대', '핵심정보', '투자', '수익률', '계좌']
    for phrase in spam_phrases:
        if phrase in text:
            return True

    return False

File: wrapped_kotlin_ported/funcs/f_70.kt
fun isSpam(text: String): Boolean {
    // Check for unusual numeric or special characters percentage
    val nonAlphabeticChars = text.count { !it.isLetterOrDigit() }
    val percentage = nonAlphabeticChars.toDouble() / text.length
    if (percentage > 0.3) {
        return true
    }

    // Check for excessively long alphanumeric strings (potential URLs)
    val alphanumericChunks = text.split(Regex("\\s+"))
    for (chunk in alphanumericChunks) {
        if (chunk.length > 20) {
            return true
        }
    }

    // Check for common spam phrases
    val spamPhrases = listOf("상한가", "최고이자율", "특별정보", "M반도체", "적금", "출금", "출시", "이벤트",
                    "공개", "혜택", "우대", "핵심정보", "투자", "수익률", "계좌")
    for (phrase in spamPhrases) {
        if (text.contains(phrase)) {
            return true
        }
    }

    return false
}
